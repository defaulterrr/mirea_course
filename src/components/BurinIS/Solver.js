import React from 'react';
import calculate from './calculate';
import './Solver.css';

class Solver extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      aCoef: '',
      bCoef: '',
      cCoef: '',
    };
    this.find = this.find.bind(this);
    this.aCoefC = this.aCoefC.bind(this);
    this.bCoefC = this.bCoefC.bind(this);
    this.cCoefC = this.cCoefC.bind(this);
  }

  find() {
    this.setState(
      (state) => ({
        solve: calculate(state.aCoef, state.bCoef, state.cCoef),
      }),
    );
  }

  aCoefC(event) {
    this.setState({ aCoef: event.target.value });
    this.find();
  }

  bCoefC(event) {
    this.setState({ bCoef: event.target.value });
    this.find();
  }

  cCoefC(event) {
    this.setState({ cCoef: event.target.value });
    this.find();
  }

  render() {
    const { solve } = this.state;
    const { aCoef } = this.state;
    const { bCoef } = this.state;
    const { cCoef } = this.state;
    return (
      <div className="input">
        <input type="number" id="aCoef" onInput={this.aCoefC} value={aCoef} />
        <h4> *x^2+ </h4>
        <input type="number" id="bCoef" onChange={this.bCoefC} value={bCoef} />
        <h4> *x+ </h4>
        <input type="number" id="cCoef" onChange={this.cCoefC} value={cCoef} />
        <h4> =0 </h4>
        <div id="solve" className="solve">
          <h4>Корни: </h4>
          {solve}
        </div>
      </div>
    );
  }
}
export default Solver;
