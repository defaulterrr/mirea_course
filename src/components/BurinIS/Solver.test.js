import React from 'react';
import { create } from 'react-test-renderer';

import Solver from './Solver';

describe('[RENDER] Solver - Snapshot test', () => {
  test('Matches the snapshot', () => {
    const captcha = create(<Solver />);
    expect(captcha.toJSON()).toMatchSnapshot();
  });
});
