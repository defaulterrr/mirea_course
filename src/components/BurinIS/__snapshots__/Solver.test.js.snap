// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[RENDER] Solver - Snapshot test Matches the snapshot 1`] = `
<div
  className="input"
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <input
    id="aCoef"
    onInput={[Function]}
    type="number"
    value=""
  />
  <h4
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
     *x^2+ 
  </h4>
  <input
    id="bCoef"
    onChange={[Function]}
    type="number"
    value=""
  />
  <h4
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
     *x+ 
  </h4>
  <input
    id="cCoef"
    onChange={[Function]}
    type="number"
    value=""
  />
  <h4
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
     =0 
  </h4>
  <div
    className="solve"
    id="solve"
  >
    <h4>
      Корни: 
    </h4>
  </div>
</div>
`;
